src/gpu/generic/sycl/ref_softmax.cpp

```diff
@@ -52,6 +52,8 @@ status_t ref_sycl_softmax_fwd_t::init(impl::engine_t *engine) {
 }
 
 status_t ref_sycl_softmax_fwd_t::execute_forward(const exec_ctx_t &ctx) const {
+    if (pd()->has_zero_dim_memory()) return status::success;
```

Comment:
Could you clarify what failure do these checks prevents?
I would expect this kind of check to happen during pd creation time.